crypto/storage/client.go

```diff
@@ -0,0 +1,109 @@
+package storage
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/crypto/storage/httpclient"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+	"net/http"
+)
+
+// APIClient implements the Storage interface. It uses a simple HTTP protocol to connect to an external storage server.
+// This server can either be a secret store itself, or proxy the request to a key store such as Hashicorp Vault or Azure Key Vault.
+// It allows us to keep the codebase clean and allow other parties to write their own adaptor.
+type APIClient struct {
+	httpClient *httpclient.ClientWithResponses
+}
+
+type APIClientConfig struct {
+	URL string `koanf:"url"`
+}
+
+func NewAPIClient(url string) (Storage, error) {
+	client, _ := httpclient.NewClientWithResponses(url)
```

Comment:
Tight time-outs are important here, we shouldn't block for seconds when just signing a transaction. Implementations should respond really, really quick (I'd say <100ms or even less), since network message handling is synchronously calling this, blocking when reading a private key (when it receives any private TX, it tries to decrypt the PAL header with all of its private keys, to find out whether its meant for the local node).